### PR TITLE
Estimate tx gas

### PR DESCRIPTION
**Pull request by cag**
Edit notes:

Discovered that `eth_call` is totally unspecified for error cases, which is why the `--noVMErrorsOnRPCResponse` got set on Ganache. Here's a summary of what is happening:

- Regular Ganache runs return a VM error with `eth_call`, and Ganache will attempt to decode the revert reason data as a UTF-8 string to send as part of the error. There is raw "return" data, <s>but I think it is actually just the original `calldata` echoed back</s> and it's an ABI-encoded Error.
- Ganache using `--noVMErrorsOnRPCResponse` and Geth will return the data supplied to `revert` call _as_ the response. There's no distinguishing this from a regular return. Our use case depends on this behavior, as buggy as it feels to me.
- Parity/OpenEthereum will return a "VM execution error." Similar to the regular Ganache, there's raw "return" data <s>which is actually just the `calldata` echoed back, so there isn't a way to use Parity to do a gas estimate through the `requiredTxGas` method</s>.

Edit 2: Found that it's not the `calldata` echoed back, but actually ABI-encoded `Error(string)` data: https://solidity.readthedocs.io/en/latest/control-structures.html#revert

This also explains the `.substring(138)` that I didn't understand.

In any case, it is a matter of getting that data now.

Edit 3: works with OpenEthereum now. Tested with the following:

```sh
parity --config dev --unlock 0x00a329c0648769a73afac7f9381e08fb43dbea72 --password <(echo '')
```

### CODE REVIEW

**Comment by cag** [comment]
Reference to batching transactions issue: https://github.com/ethers-io/ethers.js/issues/62

File: src/providers/CPKEthersProvider.ts [hunk 1]
```diff
@@ -192,6 +237,17 @@ class CPKEthersProvider implements CPKProvider {
   getSendOptions(options: object, ownerAccount: string): object {
     return options;
   }
+
+  async getGasPrice(): Promise<number> {
```
Comment (germartinez) [inline src/providers/CPKEthersProvider.ts:241]: No longer needed

File: src/providers/CPKWeb3Provider.ts [hunk 2]
```diff
@@ -173,6 +266,15 @@ class CPKWeb3Provider implements CPKProvider {
       ...(options || {}),
     };
   }
+
+  getGasPrice() {
```
Comment (germartinez) [inline src/providers/CPKWeb3Provider.ts:270]: No longer needed

**Comment by germartinez** [comment]
I just saw that this is fixed in the next PR

**Pull request merged**